src/core/utils/Counter.hpp

```diff
@@ -8,10 +8,7 @@ template <typename T> class Counter {
 private:
   T m_val;
   T m_initial;
-
-private:
   friend class boost::serialization::access;
```

Comment:
If the method is no longer private the fried declaration can be removed too.

Reply:
it's actually still private, I just removed a redundant ```private```